upup/pkg/fi/cloudup/dns.go

```diff
@@ -170,17 +174,39 @@ func precreateDNS(cluster *api.Cluster, cloud fi.Cloud) error {
 
 	for _, dnsHostname := range dnsHostnames {
 		dnsHostname = dns.EnsureDotSuffix(dnsHostname)
-		dnsRecord := recordsMap["A::"+dnsHostname]
 		found := false
-		if dnsRecord != nil {
-			rrdatas := dnsRecord.Rrdatas()
-			if len(rrdatas) > 0 {
-				glog.V(4).Infof("Found DNS record %s => %s; won't create", dnsHostname, rrdatas)
-				found = true
-			} else {
-				// This is probably an alias target; leave it alone...
-				glog.V(4).Infof("Found DNS record %s, but no records", dnsHostname)
-				found = true
+		if cloud.ProviderID() != "vsphere" {
+			dnsRecord := recordsMap["A::"+dnsHostname]
+			if dnsRecord != nil {
+				rrdatas := dnsRecord.Rrdatas()
+				if len(rrdatas) > 0 {
+					glog.V(4).Infof("Found DNS record %s => %s; won't create", dnsHostname, rrdatas)
+					found = true
+				} else {
+					// This is probably an alias target; leave it alone...
+					glog.V(4).Infof("Found DNS record %s, but no records", dnsHostname)
+					found = true
+				}
+			}
+		} else {
+			dnsRecord, err := rrs.Get(dnsHostname)
+			if err != nil {
+				return fmt.Errorf("Failed to get DNS record %s with error: %v", dnsHostname, err)
+			}
+			if dnsRecord != nil {
+				if dnsRecord.Type() != "A" {
+					glog.V(4).Infof("Found DNS record %s with type %s, continue to create A type", dnsHostname, dnsRecord.Type())
+				} else {
+					rrdatas := dnsRecord.Rrdatas()
+					if len(rrdatas) > 0 {
+						glog.V(4).Infof("Found DNS record %s => %s; won't create", dnsHostname, rrdatas)
+						found = true
+					} else {
+						// This is probably an alias target; leave it alone...
+						glog.V(4).Infof("Found DNS record %s, but no records", dnsHostname)
```

Comment:
I just copied the same logging message as what was done in the original code...
I think this means there is already a record in the server, but without a mapping value for it.
So we don't need to create a record for it (and we cannot too, because that will cause error). Later the master will just fill in the value (IP).